core/vtk/ttkIntegralLines/ttkIntegralLines.cpp

```diff
@@ -233,132 +116,119 @@ int ttkIntegralLines::getTrajectories(vtkDataSet *input,
   return 0;
 }
 
-template <typename VTK_TT>
-int ttkIntegralLines::dispatch() {
+template <typename VTK_TT, typename TTK_TT>
+int ttkIntegralLines::dispatch(int inputOffsetsDataType,
+                               const TTK_TT *triangulation) {
   int ret = 0;
-  if(inputOffsets_->GetDataType() == VTK_INT) {
-    ret = integralLines_.execute<VTK_TT, int>();
+  if(inputOffsetsDataType == VTK_INT) {
+    ret = this->execute<VTK_TT, int, TTK_TT>(triangulation);
   }
-  if(inputOffsets_->GetDataType() == VTK_ID_TYPE) {
-    ret = integralLines_.execute<VTK_TT, vtkIdType>();
+  if(inputOffsetsDataType == VTK_ID_TYPE) {
+    ret = this->execute<VTK_TT, vtkIdType, TTK_TT>(triangulation);
   }
   return ret;
 }
 
-int ttkIntegralLines::doIt(vector<vtkDataSet *> &inputs,
-                           vector<vtkDataSet *> &outputs) {
-  //                            vtkPointSet* seeds, vtkUnstructuredGrid*
-  //                            output){
+int ttkIntegralLines::RequestData(vtkInformation *request,
+                                  vtkInformationVector **inputVector,
+                                  vtkInformationVector *outputVector) {
+  vtkDataSet *domain = vtkDataSet::GetData(inputVector[0], 0);
+  vtkPointSet *seeds = vtkPointSet::GetData(inputVector[1], 0);
+  vtkUnstructuredGrid *output = vtkUnstructuredGrid::GetData(outputVector, 0);
+
+  ttk::Triangulation *triangulation = ttkAlgorithm::GetTriangulation(domain);
+  vtkDataArray *inputScalars = this->GetInputArrayToProcess(0, domain);
+
+  vtkDataArray *inputOffsets = nullptr;
+  if(this->GetInputArrayInformation(1))
```

Comment:
I actually updated the example for offsets.
if a second array was indeed provided, let's consider that the users know what they're doing and that they want to use that second array.
in that case, if that IF condition is true, thanks for also doing:
```
ForceInputOffsetScalarField = true;
```

Reply:
Would the correct logic be this?
```cpp
  vtkDataArray *inputOffsets = nullptr;
  if(this->GetInputArrayInformation(1)) {
    inputOffsets = this->GetInputArrayToProcess(1, domain);
  } else if(!ForceInputOffsetScalarField) {
    inputOffsets = domain->GetPointData()->GetArray(ttk::OffsetScalarFieldName);
  }
```
Setting `ForceInputOffsetScalarField = true` would have unexpected side effects, because that parameter is exposed to ParaView GUI, which will not be notified about the value being changed, so GUI and internal state would desynchronize.